File: level1/문자열_내_마음대로_정렬하기/solution.js
function solution(strings, n) {
  var answer = [];
  return strings.sort((s1, s2) => {
    var r = 0;
    if (s1[n] === s2[n]) r = s1 === s2 ? 0 : s1 > s2 ? 1 : -1;
    else r = s1[n] === s2[n] ? 0 : s1[n] > s2[n] ? 1 : -1;
    return r;
  });
}

console.log(solution([sun, bed, car], 1));
console.log(solution([abce, abcd, cdx], 2));

File: level1/문자열_내_마음대로_정렬하기/solution.rb
def solution(strings, n)
  answer = strings.sort { |a, b|
    v = a[n] <=> b[n]
    v.zero? ? a <=> b : v
  }
  return answer
end

p solution ["sun", "bed", "car"], 1
p solution ["abce", "abcd", "cdx"], 2
